exercises/practice/list-ops/.meta/proof.ci.ts

```diff
@@ -38,27 +38,27 @@ const Null: Cons = {
   ): TReturn {
     return initial as TReturn
   },
-  filter(): Cons {
+  filter(): Cons<undefined> {
     return Null
   },
-  reverse(): Cons {
+  reverse(): Cons<undefined> {
     return this
   },
-  map(): Cons {
-    return this
+  map<TReturn>(): Cons<TReturn> {
+    return this as Cons<TReturn>
   },
 }
-class Cons {
+class Cons<T> {
   constructor(
-    public readonly value: unknown,
-    public next: Cons = Null
+    public readonly value: T,
+    public next: Cons<T> = Null as Cons<T>
```

Comment:
It's probably better to define it as:

```typescript
public next: Cons<T> | Cons<undefined>
```

instead of needing the `as Cons<T>`

Reply:
This can make the code a little bit messier with the types, and the only method where this will be important is `get`, so that's why I made the return value of `get` be `T | undefined`